src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs

```diff
@@ -136,6 +137,11 @@ internal bool CanGenerateLocal()
                 return !this.IsInMemberContext && this.IsInExecutableBlock;
             }
 
+            internal bool CanGenerateParameter()
+            {
+                return !this.IsInMemberContext && this.ContainingMethod != null && !this.IsConstant;
```

Comment:
is the first check necessary?  if you have a containing method, is that not sufficient?  if not, doc ths line with why this check is appropriate.